tendrl/monitoring_integration/sync/dashbaord_sync.py

```diff
@@ -47,118 +63,118 @@ def refresh_dashboard(self):
-                                    dashboard_name,
-                                    cluster_details["integration_id"],
-                                    cluster_details["sds_name"]
-                                )
-                    except (KeyError, AttributeError):
-                        alert_utils.delete_alert_dashboard(dashboard_name)
-                        self.create_dashboard(
+                    else:
+                        resource_json = self.update_dashboard(
                             dashboard_name,
-                            cluster_details
+                            cluster_details[dashboard_name],
+                            resource_json
+                        )
+                except (KeyError, AttributeError):
+                    alert_utils.delete_alert_dashboard(dashboard_name)
+                    if cluster_details[dashboard_name]:
+                        resource_json = self.create_dashboard(
+                            dashboard_name,
+                            cluster_details[dashboard_name]
                         )
 
-                else:
-                    self.create_dashboard(
+            else:
+                if cluster_details[dashboard_name]:
+                    resource_json = self.create_dashboard(
                         dashboard_name,
-                        cluster_details
+                        cluster_details[dashboard_name]
                     )
+            resp = alert_utils.post_dashboard(resource_json)
+            self.log_message(resp, dashboard_name)
 
-    def create_dashboard(self, resource_name, cluster_details):
-        flag = False
-        for resource in cluster_details[resource_name]:
-            if not flag:
-                # create dashboard
-                alert_dashboard.create_resource_dashboard(
-                    resource_name,
-                    resource,
-                    cluster_details["sds_name"],
-                    cluster_details["integration_id"]
-                )
-                flag = True
-            else:
-                # update dashboard
-                self.create_gluster_resource(
-                    resource,
-                    resource_name,
-                    cluster_details["integration_id"],
-                    cluster_details["sds_name"]
-                )
+    def create_dashboard(self, resource_type, resources):
+        resource_json = {}
+        # create dashboard
+        if resources:
+            # Create operation takes more excution time than update
+            # Create dashboard for one resource using that do update for
+            # other resources
+            resource = resources.pop()
+            resource_json = alert_dashboard.create_resource_dashboard(
+                resource_type,
+                resource,
+            )
+        # After creating first panel update the other panels
+        if resources:
```

Comment:
Why this check again, we can have all the code below if condition at line#122

Reply:
only once i am allowing create a dashboard, so if any resource after creates also then i doing update dashbaord.

so let's take example only one resource in a list, so i am pop it from list and calling create a dashbaord function using that resource, now the list is empty so no need to call update dashbaord so in this check i am checking again after pop any resource is empty or has some resource